### ISSUE TITLE
`reportUnsafeMultipleInheritance` false positive on class when an `__init__` method is present in instance of the metaclass that uses `dataclass_transform`

### ISSUE THREAD

**Issue by DetachHead** [DetachHead/basedpyright#869]
Code sample in [basedpyright playground](https://basedpyright.com/?typeCheckingMode=all&code=GYJw9gtgBALgngBwJYDsDmUkQWEMoAmAhjEQMYA2RAztQPowhErXC4QCwAUN9wALFSlGvUbNW7ABQBKbsNpQAgpPgIAptIBcUAHR7ePLvOpQAYmDCSIaoVVoBeRVu5RXhNcCh06qJDG%2BS1GoUwNJQALQAfFAAcmAoapoubikIIKgwkgBEZEQUFGoEWbKGcnYmAELaejoGZSJQFUQgkuZgADSNWrp6rgDEUGog4CDaIGo4eACqLETAagCyAK4UMEgIBQCSKAAWQ37MZGp1XGkZkk0t0mFQA2coMCY5eQVFQA)

```python
from typing import dataclass_transform


@dataclass_transform()
class A(type): ...


class Foo(metaclass=A):
    def __init__(self) -> None:
        print("called")


class B: ...


class Bar(Foo, B): ...  # error: reportUnsafeMultipleInheritance


print(Bar())  # prints "called"
```

if the `__init__` was in `B` instead of `Foo`, the error would be correct. see https://github.com/DetachHead/basedpyright/issues/868#issuecomment-2472205770

**Comment by DetachHead** [comment]
actually this is just the same issue as https://github.com/DetachHead/basedpyright/issues/868#issuecomment-2472205770, since `Foo` is treated as a dataclass as well
